src/language-server/hover-documenter.ts

```diff
@@ -94,10 +95,19 @@ export default class HoverDocumenter extends Handler {
       return;
     }
     this.logger.log(`Found hover description: ${description}`);
-    // TODO(rictic): we can get ranges more often here. The tricky thing is
-    // that we want the location of the area we're mousing over, which usually
-    // isn't `feature`. We want the range of the referencer, not the referent.
-    return {contents: description, range: undefined};
+    let range = undefined;
+    if (location.language === 'html') {
+      const parsedDoc = location.document.parsedDocument as ParsedHtmlDocument;
```

Comment:
why do you need this cast?

Reply:
We don't have subtypes of `Document` for the types of their parsed documents. We might should though, now that type parameters can have default values…